### PR TITLE
Feat/get active submission

### PR DESCRIPTION
**Pull request by leoraba**
Implementation of GET `/submission/category/{categoryId}` endpoint

### CODE REVIEW

**Comment by Buwujiu** [comment]
waiting for approval on refactor PR: https://app.zenhub.com/workspaces/pcgl---pancanadian-genome-library-659eed8bc4fe0c04890cff77/issues/gh/overture-stack/lyric/19

File: packages/common/src/services/dictionaryService.ts
Comment (leoraba) [inline packages/common/src/services/dictionaryService.ts:32]: This update was left in TO DO in previous commits but implemented until now. This is not related to this PR

File: apps/server/src/routes/swagger/schemas.yml
Comment (joneubank) [inline apps/server/src/routes/swagger/schemas.yml:120]: User name or ID? Could be more explicit here.

File: apps/server/src/routes/swagger/schemas.yml
Comment (joneubank) [inline apps/server/src/routes/swagger/schemas.yml:134]: Do we need the category ID? Just the name should be enough information to work with... unless we have APIs that accept category ID.

File: apps/server/src/routes/swagger/schemas.yml
Comment (joneubank) [inline apps/server/src/routes/swagger/schemas.yml:87]: why is this organized with `additionalProperties`, seems uneccesary?

File: packages/common/src/controllers/submissionController.ts
Comment (joneubank) [inline packages/common/src/controllers/submissionController.ts:86]: should probably call this `getActive`

File: packages/common/src/external/lecternClient.ts
Comment (joneubank) [inline packages/common/src/external/lecternClient.ts:25]: If you move this inside the try block then we don't need the mutable `newSchema` variable.

File: packages/common/src/repository/activeSubmissionRepository.ts
Comment (joneubank) [inline packages/common/src/repository/activeSubmissionRepository.ts:61]: Can we only have one active submission per category? I would think we could have multiple, one per user...

File: packages/common/src/services/submissionService.ts
Comment (joneubank) [inline packages/common/src/services/submissionService.ts:146]: should probably be called `getActiveSubmission` or similar.

**Review by joneubank — CHANGES_REQUESTED** [review]
Mostly little changes around function naming and swagger descriptions...

I am concerned that we are setting this up to have only a single submission per category, when at very least we want each organization to be able to create a submission per category.

File: apps/server/src/routes/swagger/schemas.yml
Comment (leoraba) [reply apps/server/src/routes/swagger/schemas.yml:120]: Adding `user name` in the description as it is more useful

File: apps/server/src/routes/swagger/schemas.yml
Comment (leoraba) [reply apps/server/src/routes/swagger/schemas.yml:134]: There are APIs that accept category Id. For this reason we're returning the Category ID and also the Category name to identify it more easily.

File: apps/server/src/routes/swagger/schemas.yml
Comment (leoraba) [reply apps/server/src/routes/swagger/schemas.yml:87]: we use `additionalProperties` to refer to property names that are unknown, these properties or "keys" must be string an unique. It represents the `entity name`/`schema name` the records belong to.

File: packages/common/src/controllers/submissionController.ts
Comment (leoraba) [reply packages/common/src/controllers/submissionController.ts:86]: Agree. thanks for noticing

File: packages/common/src/external/lecternClient.ts
Comment (leoraba) [reply packages/common/src/external/lecternClient.ts:25]: Left outside of the try block as we want to identify 2 types of error. 
1. The try/catch block throws a ServiceUnavailable when Lectern service is down. 
2. Throw a BadRequest in case dictionary name/version doesn't exist in Lectern.

**Review by justincorrigible — COMMENTED** [review]
couple changes for consistency

File: packages/common/src/repository/activeSubmissionRepository.ts
Comment (leoraba) [reply packages/common/src/repository/activeSubmissionRepository.ts:61]: this is an important point I missed. We should be able to handle multiple submissions per category at least one per user. Will add userName param.

File: packages/common/src/services/submissionService.ts
Comment (leoraba) [reply packages/common/src/services/submissionService.ts:146]: renamed to `getActiveSubmission`

**Comment by leoraba** [comment]
Changes applied based on feedback

File: packages/common/src/external/lecternClient.ts
Comment (joneubank) [reply packages/common/src/external/lecternClient.ts:25]: Also since this function is throwing errors I checked for uses of it to make sure they are being properly wrapped in try/catch blocks... there is a usage in register dictionary without a try/catch, this means we could be throwing a `ServiceUnavailable` error from our controller. Do we have a clear response for this type in our error handler?

File: packages/common/src/controllers/submissionController.ts
Comment (joneubank) [inline packages/common/src/controllers/submissionController.ts:24]: Do we have an issue opened to plan this work?

Comment (leoraba) [reply packages/common/src/controllers/submissionController.ts:24]: will create a ticket to implement Auth

**Comment by leoraba** [comment]
Adding 3 endpoints to retrieve Active Submissions:
1. `GET /submission/category/{categoryId}` Returns a summary of all Active Submissions by categoryId and user
2. `GET /submission/category/{categoryId}/organization/{organization}` Returns a summary of all Active Submissions by categoryId, organization and user
3. `GET /submission/{submissionId}` Returns an Active Submission

File: apps/server/src/routes/swagger/submission-api.yml
Comment (joneubank) [inline apps/server/src/routes/swagger/submission-api.yml:30]: Perhaps this is better described as `List active submissions for a category` since it doesn't get the entire active submission.

File: packages/common/src/repository/activeSubmissionRepository.ts
Comment (joneubank) [inline packages/common/src/repository/activeSubmissionRepository.ts:112]: If none are found, could this just return an empty array instead of undefined?

**Review by joneubank — APPROVED** [review]

File: apps/server/src/routes/swagger/submission-api.yml
Comment (leoraba) [reply apps/server/src/routes/swagger/submission-api.yml]: Changes as suggested. Thanks.

**Pull request merged**
